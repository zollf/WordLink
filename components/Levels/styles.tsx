import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  levels: {
    marginTop: 20,
    borderRadius: 5,
    display: 'flex',
    padding: 10,
    height: '100%',
  },
  group: {
    display: 'flex',
    flexDirection: 'column',
    flexWrap: 'wrap',
  },
});

export default styles;
