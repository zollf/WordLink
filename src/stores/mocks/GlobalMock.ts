const GlobalMock: Global = {
  _ready: true,
  currentPage: 'index',
  setCurrentPage: jest.fn(),
  previousPage: 'index',
  currentDifficultyOpen: '',
  setCurrentDifficultyOpen: jest.fn((d: string) => (GlobalMock.currentDifficultyOpen = d)),
  getDifficultyColor: jest.fn(() => ''),
  setStartingPage: jest.fn(),
  onStartingPage: false,
  setUserInfo: jest.fn(),
  userInfo: {
    username: 'jeff',
    points: 20,
    puzzles: {
      completed: 2,
      easy: [],
      medium: [],
      hard: [],
    },
    hintsUsed: 6,
    hardestDifficulty: 'Medium',
    starsEarnt: 69,
  },
  init: jest.fn(),
};

export default GlobalMock;
