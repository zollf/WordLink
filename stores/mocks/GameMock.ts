const level: Level = {
  title: 'hello',
  difficulty: 'easy',
  game: {
    start: 'FORTY',
    end: 'FIFTY',
    path: 5,
  },
};

const GameMocks = {
  gameId: 0,
  difficulty: 'EASY',
  inGame: false,
  loaded: false,
  error: false,
  game: level.game,
  gameTitle: 'hello',

  currentWord: level.game.start,
  selectedLetter: '',
  stepNumber: 0,
  path: [level.game.start],
  visited: { FORTY: true },
  completed: false,

  loadGame: jest.fn(),
  startGame: jest.fn(),
  getNewLetters: jest.fn(() => ['A', 'B', 'C']),
  isWord: jest.fn(),
  setSelectedLetter: jest.fn(),
  changeLetter: jest.fn(),
  resetCurrentWord: jest.fn(),
  undo: jest.fn(),
  clearGame: jest.fn(),
};

export default GameMocks;
