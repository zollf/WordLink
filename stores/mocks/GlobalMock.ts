const GlobalMock: Global = {
  currentPage: 'index',
  setCurrentPage: jest.fn(),
  previousPage: 'index',
  currentDifficultyOpen: '',
  setCurrentDifficultyOpen: jest.fn((d: string) => (GlobalMock.currentDifficultyOpen = d)),
  getDifficultyColor: jest.fn(() => ''),
  setStartingPage: jest.fn(),
  onStartingPage: false,
};

export default GlobalMock;
