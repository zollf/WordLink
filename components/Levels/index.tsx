import React, { useEffect } from 'react';
import { View, Animated, ScrollView } from 'react-native';
import { observer } from 'mobx-react';
import { useStore, useMultiAnimate } from '../../hooks';
import { getLevel } from '../../data';
import LevelCard from '../LevelCard';

import { ScaleSlideIn, Animate } from '../../animations';
import styles from './styles';

const Levels = () => {
  const { currentDifficultyOpen } = useStore('global');
  const level = getLevel(currentDifficultyOpen);
  const animations = useMultiAnimate(0.8, Object.keys(level || {}).length);

  useEffect(() => {
    Animated.sequence([
      Animated.delay(100),
      Animated.stagger(50, 
        animations.map((a: Animated.Value) => ScaleSlideIn(a))
      ),
    ]).start();
  });

  return (
    <View style={styles.levels}>
      <ScrollView style={styles.group}>
        {level && Object.keys(level).map((key: string, i: number) => (
          <Animate scale={animations[i]} key={key}>
            <LevelCard level={level[key]} difficulty={currentDifficultyOpen} number={key}/>
          </Animate>
        ))}
      </ScrollView>
    </View>
  );
};

export default observer(Levels);