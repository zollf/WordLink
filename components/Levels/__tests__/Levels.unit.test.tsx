import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'mobx-react';
import GlobalMock from 'app/stores/mocks/GlobalMock';
import GameMock from 'app/stores/mocks/GameMock';

import Levels from '../';

GlobalMock.currentDifficultyOpen = 'easy';

const component = (
  <Provider global={GlobalMock} gameStore={GameMock}>
    <Levels />
  </Provider>
);

describe('<Levels />', () => {
  let wrapper: renderer.ReactTestRenderer;
  beforeEach(() => {
    wrapper = renderer.create(component);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('click works correctly', () => {
    wrapper.root.findAllByProps({ 'data-test-id': 'level-card' })[0].props.onPress();
    expect(GlobalMock.setCurrentPage).toBeCalled();
    expect(GameMock.loadGame).toBeCalled();
  });
});
