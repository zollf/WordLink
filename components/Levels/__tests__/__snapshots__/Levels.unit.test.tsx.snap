// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Levels /> matches snapshot 1`] = `
<View
  style={
    Object {
      "borderRadius": 5,
      "display": "flex",
      "height": "100%",
      "marginTop": 20,
      "padding": 10,
    }
  }
>
  <RCTScrollView
    showsVerticalScrollIndicator={false}
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "flexWrap": "wrap",
      }
    }
  >
    <View>
      <View
        style={
          Array [
            undefined,
            Object {
              "opacity": 1,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "scale": mockConstructor {},
                },
              ],
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <ViewManagerAdapter_ExpoLinearGradient
            proxiedProperties={
              Object {
                "colors": Array [
                  4294177779,
                  4293125091,
                ],
                "endPoint": undefined,
                "locations": Array [
                  0.5,
                  0.5,
                ],
                "startPoint": Array [
                  -0.1,
                  0.1,
                ],
              }
            }
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#E3E3E3",
                "borderRadius": 5,
                "display": "flex",
                "height": 267,
                "justifyContent": "center",
                "margin": 20,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 2,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 4,
                "width": 600,
              }
            }
          >
            <ViewManagerAdapter_ExpoLinearGradient
              proxiedProperties={
                Object {
                  "colors": Array [
                    4282711882,
                    4280928549,
                  ],
                  "endPoint": undefined,
                  "locations": undefined,
                  "startPoint": Array [
                    -0.25,
                    0.25,
                  ],
                }
              }
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "black",
                  "borderRadius": 94,
                  "bottom": "22%",
                  "display": "flex",
                  "height": 187.5,
                  "justifyContent": "center",
                  "left": "8%",
                  "position": "absolute",
                  "width": 187.5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontFamily": "Raleway_700Bold",
                    "fontSize": 66.5,
                  }
                }
              >
                1
              </Text>
            </ViewManagerAdapter_ExpoLinearGradient>
            <View
              style={
                Object {
                  "bottom": "35%",
                  "left": "40%",
                  "position": "absolute",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontFamily": "Raleway_700Bold",
                    "fontSize": 40,
                  }
                }
              >
                FORTY
              </Text>
              <Text
                style={
                  Object {
                    "fontFamily": "Raleway_700Bold",
                    "fontSize": 40,
                  }
                }
              >
                FIFTY
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "borderBottomLeftRadius": 0,
                    "borderBottomRightRadius": 0,
                    "borderRadius": 187.5,
                    "borderTopRightRadius": 0,
                    "bottom": -187.5,
                    "height": 375,
                    "position": "absolute",
                    "right": -187.5,
                    "width": 375,
                  },
                  Object {
                    "backgroundColor": "#29C925",
                  },
                ]
              }
            />
            <View
              style={
                Object {
                  "bottom": "8%",
                  "display": "flex",
                  "flexDirection": "row",
                  "left": "11%",
                  "position": "absolute",
                }
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "height": 37.5,
                    "marginLeft": "3%",
                    "marginRight": "3%",
                    "width": 37.5,
                  }
                }
              />
              <Image
                source={1}
                style={
                  Object {
                    "height": 37.5,
                    "marginLeft": "3%",
                    "marginRight": "3%",
                    "width": 37.5,
                  }
                }
              />
              <Image
                source={1}
                style={
                  Object {
                    "height": 37.5,
                    "marginLeft": "3%",
                    "marginRight": "3%",
                    "width": 37.5,
                  }
                }
              />
            </View>
          </ViewManagerAdapter_ExpoLinearGradient>
        </View>
      </View>
      <View
        style={
          Array [
            undefined,
            Object {
              "opacity": 1,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "scale": mockConstructor {},
                },
              ],
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <ViewManagerAdapter_ExpoLinearGradient
            proxiedProperties={
              Object {
                "colors": Array [
                  4294177779,
                  4293125091,
                ],
                "endPoint": undefined,
                "locations": Array [
                  0.5,
                  0.5,
                ],
                "startPoint": Array [
                  -0.1,
                  0.1,
                ],
              }
            }
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#E3E3E3",
                "borderRadius": 5,
                "display": "flex",
                "height": 267,
                "justifyContent": "center",
                "margin": 20,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 2,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 4,
                "width": 600,
              }
            }
          >
            <ViewManagerAdapter_ExpoLinearGradient
              proxiedProperties={
                Object {
                  "colors": Array [
                    4282711882,
                    4280928549,
                  ],
                  "endPoint": undefined,
                  "locations": undefined,
                  "startPoint": Array [
                    -0.25,
                    0.25,
                  ],
                }
              }
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "black",
                  "borderRadius": 94,
                  "bottom": "22%",
                  "display": "flex",
                  "height": 187.5,
                  "justifyContent": "center",
                  "left": "8%",
                  "position": "absolute",
                  "width": 187.5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontFamily": "Raleway_700Bold",
                    "fontSize": 66.5,
                  }
                }
              >
                2
              </Text>
            </ViewManagerAdapter_ExpoLinearGradient>
            <View
              style={
                Object {
                  "bottom": "35%",
                  "left": "40%",
                  "position": "absolute",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontFamily": "Raleway_700Bold",
                    "fontSize": 40,
                  }
                }
              >
                SICK
              </Text>
              <Text
                style={
                  Object {
                    "fontFamily": "Raleway_700Bold",
                    "fontSize": 40,
                  }
                }
              >
                WELL
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "borderBottomLeftRadius": 0,
                    "borderBottomRightRadius": 0,
                    "borderRadius": 187.5,
                    "borderTopRightRadius": 0,
                    "bottom": -187.5,
                    "height": 375,
                    "position": "absolute",
                    "right": -187.5,
                    "width": 375,
                  },
                  Object {
                    "backgroundColor": "#29C925",
                  },
                ]
              }
            />
            <View
              style={
                Object {
                  "bottom": "8%",
                  "display": "flex",
                  "flexDirection": "row",
                  "left": "11%",
                  "position": "absolute",
                }
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "height": 37.5,
                    "marginLeft": "3%",
                    "marginRight": "3%",
                    "width": 37.5,
                  }
                }
              />
              <Image
                source={1}
                style={
                  Object {
                    "height": 37.5,
                    "marginLeft": "3%",
                    "marginRight": "3%",
                    "width": 37.5,
                  }
                }
              />
              <Image
                source={1}
                style={
                  Object {
                    "height": 37.5,
                    "marginLeft": "3%",
                    "marginRight": "3%",
                    "width": 37.5,
                  }
                }
              />
            </View>
          </ViewManagerAdapter_ExpoLinearGradient>
        </View>
      </View>
      <View
        style={
          Array [
            undefined,
            Object {
              "opacity": 1,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "scale": mockConstructor {},
                },
              ],
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <ViewManagerAdapter_ExpoLinearGradient
            proxiedProperties={
              Object {
                "colors": Array [
                  4294177779,
                  4293125091,
                ],
                "endPoint": undefined,
                "locations": Array [
                  0.5,
                  0.5,
                ],
                "startPoint": Array [
                  -0.1,
                  0.1,
                ],
              }
            }
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#E3E3E3",
                "borderRadius": 5,
                "display": "flex",
                "height": 267,
                "justifyContent": "center",
                "margin": 20,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 2,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 4,
                "width": 600,
              }
            }
          >
            <ViewManagerAdapter_ExpoLinearGradient
              proxiedProperties={
                Object {
                  "colors": Array [
                    4282711882,
                    4280928549,
                  ],
                  "endPoint": undefined,
                  "locations": undefined,
                  "startPoint": Array [
                    -0.25,
                    0.25,
                  ],
                }
              }
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "black",
                  "borderRadius": 94,
                  "bottom": "22%",
                  "display": "flex",
                  "height": 187.5,
                  "justifyContent": "center",
                  "left": "8%",
                  "position": "absolute",
                  "width": 187.5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontFamily": "Raleway_700Bold",
                    "fontSize": 66.5,
                  }
                }
              >
                3
              </Text>
            </ViewManagerAdapter_ExpoLinearGradient>
            <View
              style={
                Object {
                  "bottom": "35%",
                  "left": "40%",
                  "position": "absolute",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontFamily": "Raleway_700Bold",
                    "fontSize": 40,
                  }
                }
              >
                RICH
              </Text>
              <Text
                style={
                  Object {
                    "fontFamily": "Raleway_700Bold",
                    "fontSize": 40,
                  }
                }
              >
                POOR
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "borderBottomLeftRadius": 0,
                    "borderBottomRightRadius": 0,
                    "borderRadius": 187.5,
                    "borderTopRightRadius": 0,
                    "bottom": -187.5,
                    "height": 375,
                    "position": "absolute",
                    "right": -187.5,
                    "width": 375,
                  },
                  Object {
                    "backgroundColor": "#29C925",
                  },
                ]
              }
            />
            <View
              style={
                Object {
                  "bottom": "8%",
                  "display": "flex",
                  "flexDirection": "row",
                  "left": "11%",
                  "position": "absolute",
                }
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "height": 37.5,
                    "marginLeft": "3%",
                    "marginRight": "3%",
                    "width": 37.5,
                  }
                }
              />
              <Image
                source={1}
                style={
                  Object {
                    "height": 37.5,
                    "marginLeft": "3%",
                    "marginRight": "3%",
                    "width": 37.5,
                  }
                }
              />
              <Image
                source={1}
                style={
                  Object {
                    "height": 37.5,
                    "marginLeft": "3%",
                    "marginRight": "3%",
                    "width": 37.5,
                  }
                }
              />
            </View>
          </ViewManagerAdapter_ExpoLinearGradient>
        </View>
      </View>
      <View
        style={
          Array [
            undefined,
            Object {
              "opacity": 1,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "scale": mockConstructor {},
                },
              ],
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <ViewManagerAdapter_ExpoLinearGradient
            proxiedProperties={
              Object {
                "colors": Array [
                  4294177779,
                  4293125091,
                ],
                "endPoint": undefined,
                "locations": Array [
                  0.5,
                  0.5,
                ],
                "startPoint": Array [
                  -0.1,
                  0.1,
                ],
              }
            }
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#E3E3E3",
                "borderRadius": 5,
                "display": "flex",
                "height": 267,
                "justifyContent": "center",
                "margin": 20,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 2,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 4,
                "width": 600,
              }
            }
          >
            <ViewManagerAdapter_ExpoLinearGradient
              proxiedProperties={
                Object {
                  "colors": Array [
                    4282711882,
                    4280928549,
                  ],
                  "endPoint": undefined,
                  "locations": undefined,
                  "startPoint": Array [
                    -0.25,
                    0.25,
                  ],
                }
              }
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "black",
                  "borderRadius": 94,
                  "bottom": "22%",
                  "display": "flex",
                  "height": 187.5,
                  "justifyContent": "center",
                  "left": "8%",
                  "position": "absolute",
                  "width": 187.5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontFamily": "Raleway_700Bold",
                    "fontSize": 66.5,
                  }
                }
              >
                4
              </Text>
            </ViewManagerAdapter_ExpoLinearGradient>
            <View
              style={
                Object {
                  "bottom": "35%",
                  "left": "40%",
                  "position": "absolute",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontFamily": "Raleway_700Bold",
                    "fontSize": 40,
                  }
                }
              >
                CENT
              </Text>
              <Text
                style={
                  Object {
                    "fontFamily": "Raleway_700Bold",
                    "fontSize": 40,
                  }
                }
              >
                DIME
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "borderBottomLeftRadius": 0,
                    "borderBottomRightRadius": 0,
                    "borderRadius": 187.5,
                    "borderTopRightRadius": 0,
                    "bottom": -187.5,
                    "height": 375,
                    "position": "absolute",
                    "right": -187.5,
                    "width": 375,
                  },
                  Object {
                    "backgroundColor": "#29C925",
                  },
                ]
              }
            />
            <View
              style={
                Object {
                  "bottom": "8%",
                  "display": "flex",
                  "flexDirection": "row",
                  "left": "11%",
                  "position": "absolute",
                }
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "height": 37.5,
                    "marginLeft": "3%",
                    "marginRight": "3%",
                    "width": 37.5,
                  }
                }
              />
              <Image
                source={1}
                style={
                  Object {
                    "height": 37.5,
                    "marginLeft": "3%",
                    "marginRight": "3%",
                    "width": 37.5,
                  }
                }
              />
              <Image
                source={1}
                style={
                  Object {
                    "height": 37.5,
                    "marginLeft": "3%",
                    "marginRight": "3%",
                    "width": 37.5,
                  }
                }
              />
            </View>
          </ViewManagerAdapter_ExpoLinearGradient>
        </View>
      </View>
      <View
        style={
          Array [
            undefined,
            Object {
              "opacity": 1,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "scale": mockConstructor {},
                },
              ],
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <ViewManagerAdapter_ExpoLinearGradient
            proxiedProperties={
              Object {
                "colors": Array [
                  4294177779,
                  4293125091,
                ],
                "endPoint": undefined,
                "locations": Array [
                  0.5,
                  0.5,
                ],
                "startPoint": Array [
                  -0.1,
                  0.1,
                ],
              }
            }
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#E3E3E3",
                "borderRadius": 5,
                "display": "flex",
                "height": 267,
                "justifyContent": "center",
                "margin": 20,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 2,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 4,
                "width": 600,
              }
            }
          >
            <ViewManagerAdapter_ExpoLinearGradient
              proxiedProperties={
                Object {
                  "colors": Array [
                    4282711882,
                    4280928549,
                  ],
                  "endPoint": undefined,
                  "locations": undefined,
                  "startPoint": Array [
                    -0.25,
                    0.25,
                  ],
                }
              }
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "black",
                  "borderRadius": 94,
                  "bottom": "22%",
                  "display": "flex",
                  "height": 187.5,
                  "justifyContent": "center",
                  "left": "8%",
                  "position": "absolute",
                  "width": 187.5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontFamily": "Raleway_700Bold",
                    "fontSize": 66.5,
                  }
                }
              >
                5
              </Text>
            </ViewManagerAdapter_ExpoLinearGradient>
            <View
              style={
                Object {
                  "bottom": "35%",
                  "left": "40%",
                  "position": "absolute",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontFamily": "Raleway_700Bold",
                    "fontSize": 40,
                  }
                }
              >
                DRY
              </Text>
              <Text
                style={
                  Object {
                    "fontFamily": "Raleway_700Bold",
                    "fontSize": 40,
                  }
                }
              >
                WET
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "borderBottomLeftRadius": 0,
                    "borderBottomRightRadius": 0,
                    "borderRadius": 187.5,
                    "borderTopRightRadius": 0,
                    "bottom": -187.5,
                    "height": 375,
                    "position": "absolute",
                    "right": -187.5,
                    "width": 375,
                  },
                  Object {
                    "backgroundColor": "#29C925",
                  },
                ]
              }
            />
            <View
              style={
                Object {
                  "bottom": "8%",
                  "display": "flex",
                  "flexDirection": "row",
                  "left": "11%",
                  "position": "absolute",
                }
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "height": 37.5,
                    "marginLeft": "3%",
                    "marginRight": "3%",
                    "width": 37.5,
                  }
                }
              />
              <Image
                source={1}
                style={
                  Object {
                    "height": 37.5,
                    "marginLeft": "3%",
                    "marginRight": "3%",
                    "width": 37.5,
                  }
                }
              />
              <Image
                source={1}
                style={
                  Object {
                    "height": 37.5,
                    "marginLeft": "3%",
                    "marginRight": "3%",
                    "width": 37.5,
                  }
                }
              />
            </View>
          </ViewManagerAdapter_ExpoLinearGradient>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
